src/main/java/org/rulez/demokracia/pdengine/BeatTable.java

```diff
@@ -0,0 +1,69 @@
+package org.rulez.demokracia.pdengine;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.rulez.demokracia.pdengine.dataobjects.Pair;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.types.MapMatrix;
+import org.rulez.demokracia.types.Matrix;
+
+public class BeatTable extends MapMatrix<Choice, Pair> implements Matrix<Choice, Pair>{
+	private static final long serialVersionUID = 1L;
+
+	public enum Direction {
+		DIRECTION_FORWARD, DIRECTION_BACKWARD
+	}
+
+	public BeatTable() {
+		this(new ArrayList<Choice>());
+	}
+
+	public BeatTable(final Collection<Choice> keyCollection) {
+		super(keyCollection);
+	}
+
+	public int beatInformation(final Choice choice1, final Choice choice2, final Direction direction) {
+		if (direction == null)
+			throw new IllegalArgumentException("Invalid direction");
+
+		int result = 0;
+		Pair pair = getPair(choice1, choice2);
+
+		if (direction.equals(Direction.DIRECTION_FORWARD))
+			result = pair.winning;
+		else
+			result = pair.losing;
+
+		return result;
+	}
+
+	public Pair getPair(final Choice choice1, final Choice choice2) {
```

Comment:
It is not a good idea to shadow a getter and setter just to delegate to the normal getter/setter.
Please unimplement them.
Yeah, I know, this was in the design. But the design is not good.

Reply:
If I unimplement the getPair how can I get the pairs in the interface?

Reply:
with getElement